quark/core/shurikenapkinfo.py

```diff
@@ -0,0 +1,532 @@
+
+    def get_method_bytecode(
+        self, method_object: MethodObject
+    ) -> Iterator[BytecodeObject]:
+        """
+        Inherited from baseapkinfo.py.
+        Return the bytecodes of the specified method.
+
+        :param method_object: a target method to get the corresponding
+        bytecodes
+        :yield: a generator of BytecodeObjects
+        """
+        disassembledMethod = self.__getDisassembledMethod(method_object)
+
+        for i in range(disassembledMethod.n_of_instructions):
+            rawBytecode = disassembledMethod.instructions[
+                i
+            ].disassembly.decode(errors="backslashreplace")
+            yield self._parseSmali(rawBytecode)
+
+    def _parseParameters(self, parameter: str) -> Union[int, float, str]:
+
+        if parameter[:2] == "0x":
+            try:
+                parameter = int(parameter, 16)
+                return parameter
+            except (TypeError, ValueError):
+                pass
+
+        try:
+            parameter = int(parameter, 10)
+            return parameter
+        except (TypeError, ValueError):
+            pass
+        try:
+            parameter = float(parameter)
+            return parameter
+        except (TypeError, ValueError):
+            pass
+
+        typeTable = {
+            "void": "V",
+            "boolean": "Z",
+            "byte": "B",
+            "short": "S",
+            "char": "C",
+            "int": "I",
+            "long": "J",
+            "float": "F",
+            "double": "D",
+        }
+        for typeName, abbreviation in typeTable.items():
+            parameter = parameter.strip()
+            pattern = r" ({})(\[\])*$".format(typeName)
+            if re.search(pattern, parameter):
+                parameter = re.sub(
+                    pattern, r" {}\2".format(abbreviation), parameter
+                )
+                break
+
+        patternToIdentifyMemberField = r"->\w+(?!\(\)) "
+        if re.search(patternToIdentifyMemberField, parameter):
+            parameter = self._convertMemberFieldFormat(parameter)
+
+        return parameter
+
+    def _parseSmali(self, smali: str) -> BytecodeObject:
+
+        smali = smali.split("//")[0].strip()
+        if smali == "":
+            raise ValueError("Argument cannot be empty.")
+
+        if " " not in smali:
+            return BytecodeObject(smali, None, None)
+
+        mnemonic, args = smali.split(maxsplit=1)
+        parameter = None
+
+        # extract string
+        quoteChar = args[-1]
+        if quoteChar == '"' or quoteChar == "'":
+
+            firstQuotePosition = args.find(quoteChar)
+            parameter = args[firstQuotePosition:][1:-1]
+            args = args[:firstQuotePosition].strip()
+
+        argsList = [arg.strip() for arg in re.split("[{},]+", args) if arg]
+
+        if parameter is None:
+            if argsList and not argsList[-1].startswith("v"):
+                parameter = self._parseParameters(argsList.pop())
```

Comment:
We can combine the two if statements to make the code simpler.

```suggestion
        if parameter is None and argsList and not argsList[-1].startswith("v"):
            # Extract literal, field, or method
            parameter = self._parseParameters(argsList.pop())
```